samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -0,0 +1,395 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.Quantum.Samples.ColoringGroverWithConstraints {
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// Read color from a register
+    ///
+    /// # Input
+    /// ## register
+    /// The qubits to measure
+    operation MeasureColor (register : Qubit[]) : Int {
+        return ResultArrayAsInt(MultiM(register));
+    }
+
+    /// # Summary
+    /// Read coloring from a register
+    ///
+    /// # Input
+    /// ## bitsPerColor
+    /// Number of bits per color
+    /// ## register
+    /// register of qubits 
+    operation MeasureColoring (bitsPerColor : Int, register : Qubit[]) : Int[] {
+        let numVertices = Length(register) / bitsPerColor;
+        let colorPartitions = Partitioned(ConstantArray(numVertices - 1, bitsPerColor), register);
+        return ForEach(MeasureColor, colorPartitions);
+    } 
+
+    /// # Summary
+    /// N-bit color equality oracle (no extra qubits)
+    ///
+    /// # Input
+    /// ## color0
+    /// First color
+    /// ## color1
+    /// Second color
+    ///
+    /// # Output
+    /// target will be 1 if colors are the same
+    operation ApplyColorEqualityOracle (color0 : Qubit[], color1 : Qubit[], 
+        target : Qubit) : Unit is Adj+Ctl {
+        within {
+            for ((q0, q1) in Zip(color0, color1)) {
+                // compute XOR of q0 and q1 in place (storing it in q1)
+                CNOT(q0, q1);
+            }
+        } apply {
+            // if all XORs are 0, the bit strings are equal
+            (ControlledOnInt(0, X))(color1, target);
+        }
+    }
+
+    /// # Summary
+    /// Oracle for verifying vertex coloring, including color constraints from
+    /// non qubit vertices
+    ///
+    /// # Input
+    /// ## numVertices
+    /// The number of vertices in the graph
+    /// ## bitsPerColor
+    /// The bits per color e.g. 2 bits per color allows for 4 colors
+    /// ## edges
+    /// The array of (Vertex#,Vertex#) specifying the Vertices that can not be
+    /// the same color
+    /// ## startingColorConstraints
+    /// The array of (Vertex#,Color) specifying the dissallowed colors for vertices
+    ///
+    /// # Output
+    /// A unitary operation that applies `oracle` on the target register if the control 
+    /// register state corresponds to the bit mask `bits`.
+    ///
+    /// # Example
+    /// Consider the following 4x4 Sudoku puzzle 
+    ///     -----------------
```

Comment:
This won't come through Markdown formatting all that great; you may want to either use GitHub-Flavored Markdown (GFM) table syntax here, or to isolate in a code fence (` ``` `).